modelgauge/main.py

```diff
@@ -280,123 +287,82 @@ def run_test(
 )
 @click.argument(
     "filename",
-    type=click.Path(exists=True),
+    type=click.Path(exists=True, path_type=pathlib.Path),
 )
-def run_prompts(sut_uids, workers, cache_dir: pathlib.Path, debug, filename):
-    """Take a CSV of prompts and run them through SUTs. The CSV file must have UID and Text columns, and may have others."""
-
+def run_prompts(sut_uids, annotator_uids, workers, sut_cache_dir, debug, filename):
```

Comment:
This function is long and hard to follow. Could it be broken up some, either by extracting functions or by creating some sort of state object or runner object?

Reply:
That's fair, it was pretty long. I just updated it. Does it look better now?

Reply:
I think this is getting closer, but I think it's still confusing. Some of what I think of as the CLI work, like validating arguments, printing user messages, and progress bars, is now off in another file. The functional decomposition with descriptive names does help convey more of the intent. But the functions have a lot of arguments, many in common. And they have a fair bit of branching, including spots with similar code, like in the `if run_annotators` block.

Instead of a functional breakdown, maybe a PipelineRunner object would help? Happy to sketch something if you'd like.

Reply:
Do you imagine `PipelineRunner` to handle all 3 types of pipelines?

Reply:
I'm thinking so. When you have a bunch of things that keep appearing in function signatures together, that's a sign that there might be an object waiting to be named, where those are the fields of the object. When there is code that goes with them, that could end up as methods on the object. Seems like there's some common stuff to all three pipelines!

Reply:
Okay, I put the logic into a runner object. What do you think?